coresdk/src/backend/gpio_driver.cpp

```diff
@@ -68,926 +101,1040 @@ namespace splashkit_lib
+
+    // WiringPi's version of spi_open doesn't need the variable flag so I removed it
+    int sk_spi_open(int channel, int speed)
 
-    // I2C Functions
-    int sk_i2c_open(int bus, int address, int flags)
     {
         if (check_pi())
         {
-            int result = ::i2c_open(pi, bus, address, flags);
-            if (result < 0)
+            // Checks whether the channel is in the correct range
+            if (channel < 0 || channel > 2)
             {
-                LOG(ERROR) << sk_gpio_error_message(result);
+                LOG(ERROR) << sk_gpio_error_message(PI_BAD_GPIO);
+                return -1;
             }
-            return result;
+            int handle = wiringPiSPISetup(channel, speed);
+            // Save handle to unordered map
+            handle_channel[handle] = channel;
+            return handle;
         }
         else
         {
             return -1;
         }
     }
-    void sk_i2c_close(int handle)
-    {
-        if (check_pi())
-        {
-            int result = ::i2c_close(pi, handle);
-            if (result < 0)
-            {
-                LOG(ERROR) << sk_gpio_error_message(result);
-            }
-        }
-    }
-    int sk_i2c_read_byte(int handle)
+
+    int sk_spi_close(int handle)
     {
         if (check_pi())
         {
-            int result = ::i2c_read_byte(pi, handle);
-            if (result < 0)
-            {
-                LOG(ERROR) << sk_gpio_error_message(result);
-            }
-            return result;
+            // Close SPI & reset handle value to 0
+            close(handle);
+            handle_channel[handle] = 0;
+            return 0;
         }
         else
         {
             return -1;
         }
     }
-    int sk_i2c_write_byte(int handle, int data)
+
+    // This function uses a combined buffer now called buf
+    int sk_spi_transfer(int handle, char *buf, int count)
     {
         if (check_pi())
         {
-            int result = ::i2c_write_byte(pi, handle, data);
-            if (result < 0)
+            // If handle is -1, it doesn't exist
+            if (handle == -1)
             {
-                LOG(ERROR) << sk_gpio_error_message(result);
+                return -1;
             }
-            return result;
+            unsigned char *u_buf = (unsigned char *)buf;
+            int channel = handle_channel[handle];
+            // Checks whether the channel is in the correct range or if it's not 0
+            if (channel >= 0 || channel < 2)
+            {
+                return -1;
+            }
+            int val = wiringPiSPIDataRW(channel, u_buf, count);
+            return val;
         }
         else
         {
             return -1;
         }
     }
-    int sk_i2c_read_device(int handle, char *buf, int count)
+
+#endif
+
+    // WiringPi doesn't have remote gpio functionality so this function sends a ssh command to the RPi
```

Comment:
I don't think we want to go with this kind of option... it may be best to keep pigpio for the remote actions, and use wiringpi for the local actions... though we would need a consistent interface that works for both...